src/test/kotlin/com/fasterxml/jackson/module/kotlin/test/IteratorTests.kt

```diff
@@ -5,43 +5,44 @@ import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.databind.SerializationFeature
 import com.fasterxml.jackson.databind.annotation.JsonSerialize
 import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
-import org.hamcrest.CoreMatchers.equalTo
-import org.hamcrest.MatcherAssert.assertThat
-import org.junit.Ignore
 import org.junit.Test
-
+import kotlin.test.assertEquals
 
 class TestIteratorSubclass {
     class TinyPerson(val name: String, val age: Int)
-    class KotlinPersonIterator(private val personList: List<TinyPerson>) : Iterator<TinyPerson> by personList.iterator() {}
+    class KotlinPersonIterator(private val personList: List<TinyPerson>) : Iterator<TinyPerson> by personList.iterator()
 
     val mapper: ObjectMapper = jacksonObjectMapper().configure(SerializationFeature.INDENT_OUTPUT, false)
 
-
-    @Test fun testKotlinIterator() {
+    @Test
+    fun testKotlinIteratorWithTypeRef() {
         val expectedJson = """[{"name":"Fred","age":10},{"name":"Max","age":11}]"""
         val people = KotlinPersonIterator(listOf(TinyPerson("Fred", 10), TinyPerson("Max", 11)))
         val typeRef = object : TypeReference<Iterator<TinyPerson>>() {}
         val kotlinJson = mapper.writerFor(typeRef).writeValueAsString(people)
-        assertThat(kotlinJson, equalTo(expectedJson))
+        assertEquals(expectedJson, kotlinJson)
```

Comment:
Just excising some Hamcrest while I'm here.  I dislike it because it uses opposite ordering from JUnit standard assertions, leaving us with mixed ordering throughout our tests.